src/validation.h

```diff
@@ -990,6 +984,15 @@ class ChainstateManager
     /** Produce the necessary coinbase commitment for a block (modifies the hash, don't call for mined blocks). */
     std::vector<unsigned char> GenerateCoinbaseCommitment(CBlock& block, const CBlockIndex* pindexPrev) const;
 
+    //! When starting up, search the datadir for a chainstate based on a UTXO
+    //! snapshot that is in the process of being validated.
+    bool DetectSnapshotChainstate(CTxMemPool* mempool) EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
+
+    void ResetChainstates() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
+
+    CChainState& ActivateExistingSnapshot(CTxMemPool* mempool, uint256 base_blockhash)
```

Comment:
even if name is explicit, could be documented "Switch active chainstate to snapshot one" in case of future refactor to check function correctness (and `DetectSnapshotChainstate` could say it activates the snapshot if any)

Reply:
Done, thanks.